DragaliaAPI/DragaliaAPI/Services/Game/StoryService.cs

```diff
@@ -256,6 +261,16 @@ private async Task<List<AtgenBuildEventRewardEntityList>> ReadQuestStory(int sto
             );
         }
 
+        if (storyId == 1001009)
+        {
+            logger.LogDebug("Granting chapter 10 completion rewards.");
+
+            IEnumerable<Present> presents = GetChapter10PresentList();
+            presentService.AddPresent(presents);
+
+            await userService.AddExperience(69990);
```

Comment:
We are not adding anything to `rewardList` here - does the client correctly show the rewards you have gained?

Reply:
You're right that they don't show in the reward list. I'll move the code up to the reward section to make sure that happens.